packages/react-native/Libraries/Image/RCTImageUtils.mm

```diff
@@ -257,59 +319,22 @@ BOOL RCTUpscalingRequired(
 UIImage *__nullable RCTDecodeImageWithData(NSData *data, CGSize destSize, CGFloat destScale, RCTResizeMode resizeMode)
 {
   CGImageSourceRef sourceRef = CGImageSourceCreateWithData((__bridge CFDataRef)data, NULL);
-  if (!sourceRef) {
-    return nil;
-  }
+    return decodeImageFromCGImageSourceRef(sourceRef, destSize, destScale, resizeMode);
 
-  // Get original image size
-  CFDictionaryRef imageProperties = CGImageSourceCopyPropertiesAtIndex(sourceRef, 0, NULL);
-  if (!imageProperties) {
-    CFRelease(sourceRef);
-    return nil;
-  }
-  NSNumber *width = (NSNumber *)CFDictionaryGetValue(imageProperties, kCGImagePropertyPixelWidth);
-  NSNumber *height = (NSNumber *)CFDictionaryGetValue(imageProperties, kCGImagePropertyPixelHeight);
-  CGSize sourceSize = {width.doubleValue, height.doubleValue};
-  CFRelease(imageProperties);
 
-  if (CGSizeEqualToSize(destSize, CGSizeZero)) {
-    destSize = sourceSize;
-    if (!destScale) {
-      destScale = 1;
-    }
-  } else if (!destScale) {
-    destScale = RCTScreenScale();
-  }
+ }
 
-  if (resizeMode == RCTResizeModeStretch) {
-    // Decoder cannot change aspect ratio, so RCTResizeModeStretch is equivalent
-    // to RCTResizeModeCover for our purposes
-    resizeMode = RCTResizeModeCover;
-  }
-
-  // Calculate target size
-  CGSize targetSize = RCTTargetSize(sourceSize, 1, destSize, destScale, resizeMode, NO);
-  CGSize targetPixelSize = RCTSizeInPixels(targetSize, destScale);
-  CGFloat maxPixelSize =
-      fmax(fmin(sourceSize.width, targetPixelSize.width), fmin(sourceSize.height, targetPixelSize.height));
+UIImage *__nullable RCTDecodeImageWithLocalAssetURL(NSURL *url, CGSize destSize, CGFloat destScale, RCTResizeMode resizeMode)
+{
+  CGImageSourceRef sourceRef = CGImageSourceCreateWithURL((__bridge CFURLRef)url, NULL);
```

Comment:
Sorry for the late reviewev. A concern has been raised internallt that this implementation might bypass `[UIImage imageNamed:]` which also works as a cache and it is more efficient for small images like icons.

A suggestion is to use `decodeImageFromCGImageSourceRef` only if the resizeMode is different from the default.

So, the logic would be (pseudocode here):
```
if (resizeMode is the default) {
  return RCTImageFromLocalAssetURL(url);
}

//rest of the current method
```

what do you think?